simpler_eda/numerical_eda.py

```diff
@@ -80,5 +74,146 @@ def numerical_plot(
+
+    # Ensure title variable is specified by the user
+    assert title is not None, "Variable title needs to be specified."
+
+    error_one = "TypeError: Data must be entered as a pandas dataframe."
+    assert isinstance(data, pd.DataFrame), error_one
+
+    error_two = "Wrong type! X-axis variable must be entered as a String."
+    assert isinstance(xval, str), error_two
+
+    error_three = "Wrong type! Y-axis variable must be entered as a String."
+    assert isinstance(yval, str), error_three
+
+    error_four = "TypeError: x_transform must be of type boolean."
+    assert isinstance(x_transform, bool), error_four
+
+    error_five = "TypeError: y_transform must be of type boolean."
+    assert isinstance(y_transform, bool), error_five
+
+    error_six = "TypeError: plot_type must be either 'scatter' or 'line'."
+    assert plot_type in ["scatter", "line"], error_six
+
+    error_seven = "TypeError: plot_width must be an integer."
+    assert isinstance(plot_width, int), error_seven
+
+    error_eight = "TypeError: plot_height must be an integer."
+    assert isinstance(plot_height, int), error_eight
+
+    error_nine = "TypeError: font_size must be a positive integer."
+    assert isinstance(font_size, int), error_nine
+
+    error_ten = "Wrong type!: color must be a string."
+    assert isinstance(color, str), error_ten
+
+    error_eleven = "Wrong type!: title must be a string."
+    assert isinstance(title, str), error_eleven
+
+    error_twelve = "Wrong type!: color_scheme must be a string."
+    assert isinstance(color_scheme, str), error_twelve
+
+    # create a copy of the dataframe so that original dataframe
+    #  remains unchanged
+    df = data.copy(deep=True)
+
+    # Ensure variable exists in the dataframe
+    assert xval in df.columns, "Variable xval not found in input dataframe."
+
+    assert yval in df.columns, "Variable yval not found in input dataframe."
+
+    # Ensure variable is numeric
+    error_msg_x = "Your x-variable needs to be numeric."
+    assert pd.api.types.is_numeric_dtype(df[xval]), error_msg_x
+
+    error_msg_y = "Your y-variable needs to be numeric."
+    assert pd.api.types.is_numeric_dtype(df[yval]), error_msg_y
+
+    # Toggle a log transformation on the x-axis
+    warn_one = "Can't have negative x values with np.log"
+    if x_transform:
+        if any(df[xval] < 0):
+            warnings.warn(warn_one)
+        df[xval] = np.log(df[xval])
+        xval = xval
+
+    # Toggle a log transformation on the y-axis
+    warn_two = "Can't have negative y values with np.log"
+    if y_transform:
+        if any(df[yval] < 0):
+            warnings.warn(warn_two)
+        df[yval] = np.log(df[yval])
+
+    # Update scale bounds of the plots
+    x_scale = alt.Scale(domain=(float(df[xval].min()), float(df[xval].max())))
+    y_scale = alt.Scale(domain=(float(df[yval].min()), float(df[yval].max())))
+
+    # Renaming column names, replacing underscores with space
```

Comment:
Are we cleaning the data frame and input from users here? What's the purpose? To make sure they are in the same format?  I might have missed something here.

Reply:
The only purpose for this is that the labels for the plot will be without underscores.